File: src/dataset/typescript/nucleotide-count_exercism-samajammin/claude/ClaudeSonnet4_nucleotide-count_v2.ts
class NucleotideCount {
  static nucleotideCounts(dna: string) {
    const nucleotideCount = { A: 0, G: 0, T: 0, C: 0 };
    
    for (let i = 0; i < dna.length; i++) {
      const nucleotide = dna[i];
      switch (nucleotide) {
        case 'A':
          nucleotideCount.A++;
          break;
        case 'G':
          nucleotideCount.G++;
          break;
        case 'T':
          nucleotideCount.T++;
          break;
        case 'C':
          nucleotideCount.C++;
          break;
        default:
          throw new Error('Invalid nucleotide in strand');
      }
    }
    
    return nucleotideCount;
  }
}

export default NucleotideCount;

File: src/dataset/cpp/nucleotide-count_exercism-kchoadley/claude/ClaudeSonnet4_nucleotide_count_v2.cpp
#include "nucleotide_count.h"
#include <stdexcept>

namespace dna {
    
    counter::counter(const std::string& strand)
    {
        nucleotides['A'] = 0;
        nucleotides['T'] = 0;
        nucleotides['C'] = 0;
        nucleotides['G'] = 0;
        
        for(char c : strand) 
        {
            switch(c) {
                case 'A':
                case 'T':
                case 'C':
                case 'G':
                    nucleotides[c]++;
                    break;
                default:
                    throw std::invalid_argument("Invalid argument");
            }
        }
    }
    
    int counter::count(const char& c) const
    {
        switch(c) {
            case 'A':
            case 'T':
            case 'C':
            case 'G':
                return nucleotides.at(c);
            default:
                throw std::invalid_argument("Invalid argument");
        }
    }
    
    const std::map<char, int>& counter::nucleotide_counts() const
    {
        return nucleotides;
    }
}